Framework/API/src/Run.cpp

```diff
@@ -56,6 +56,8 @@ Run::Run(const Run &other) : LogManager(other), m_histoBins(other.m_histoBins) {
 Run::~Run() = default;
 
 Run &Run::operator=(const Run &other) {
+  if (this == &other)
```

Comment:
What scenario justifies having self-assignment being permissive instead of throwing an error? If someone writes a self-assign statement, I would assume they're making a mistake 🤔

Reply:
It's happening in https://github.com/mantidproject/mantid/blob/master/Framework/Algorithms/src/Plus.cpp#L210

Reply:
I don't know if that is justified usage or not.

We seem to have this same check in many other `operator=` as well so it must be a common enough occurrence, _e.g._ https://github.com/mantidproject/mantid/blob/master/Framework/API/src/Sample.cpp#L59-L60